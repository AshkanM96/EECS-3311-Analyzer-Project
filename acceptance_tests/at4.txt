for_all
open_set_enumeration 
conjoin
boolean_constant (true)
boolean_constant (false)
disjoin
boolean_constant (true)
boolean_constant (false)
close_set_enumeration
type_check
evaluate 
