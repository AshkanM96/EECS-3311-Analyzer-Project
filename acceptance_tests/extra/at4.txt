open_set
open_set
bool(false)
close_set
difference
open_set
divide
int(2)
int(-0)
close_set
open_set
add
divide
int(9)
int(0)
int(132321)
close_set
close_set
evaluate
type_check
reset
open_set
difference
open_set
divide
int(2)
int(-0)
close_set
open_set
add
divide
int(9)
int(0)
int(132321)
close_set
open_set
bool(true)
close_set
open_set
bool(false)
close_set
open_set
bool(true)
close_set
open_set
bool(true)
close_set
open_set
bool(false)
close_set
open_set
bool(false)
close_set
open_set
bool(false)
close_set
close_set
evaluate
type_check
reset
