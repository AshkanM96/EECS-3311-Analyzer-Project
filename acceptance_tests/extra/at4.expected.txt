  Expression currently specified: ?
  Report: Expression is initialized.
->open_set_enumeration
  Expression currently specified: {?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {{?}}
  Report: OK.
->boolean_constant(False)
  Expression currently specified: {{False, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{False}, ?}
  Report: OK.
->difference
  Expression currently specified: {{False}, (? \ nil)}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {{False}, ({?} \ nil)}
  Report: OK.
->divide
  Expression currently specified: {{False}, ({(? / nil)} \ nil)}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {{False}, ({(2 / ?)} \ nil)}
  Report: OK.
->integer_constant(0)
  Expression currently specified: {{False}, ({(2 / 0), ?} \ nil)}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{False}, ({(2 / 0)} \ ?)}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {{False}, ({(2 / 0)} \ {?})}
  Report: OK.
->add
  Expression currently specified: {{False}, ({(2 / 0)} \ {(? + nil)})}
  Report: OK.
->divide
  Expression currently specified: {{False}, ({(2 / 0)} \ {((? / nil) + nil)})}
  Report: OK.
->integer_constant(9)
  Expression currently specified: {{False}, ({(2 / 0)} \ {((9 / ?) + nil)})}
  Report: OK.
->integer_constant(0)
  Expression currently specified: {{False}, ({(2 / 0)} \ {((9 / 0) + ?)})}
  Report: OK.
->integer_constant(132321)
  Expression currently specified: {{False}, ({(2 / 0)} \ {((9 / 0) + 132321), ?})}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{False}, ({(2 / 0)} \ {((9 / 0) + 132321)}), ?}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {{False}, ({(2 / 0)} \ {((9 / 0) + 132321)})}
  Report: OK.
->evaluate
  Expression currently specified: {{False}, ({(2 / 0)} \ {((9 / 0) + 132321)})}
  Report: Error (Divisor is zero).
->type_check
  Expression currently specified: {{False}, ({(2 / 0)} \ {((9 / 0) + 132321)})}
  Report: {{False}, ({(2 / 0)} \ {((9 / 0) + 132321)})} is type-correct.
->reset
  Expression currently specified: ?
  Report: OK.
->open_set_enumeration
  Expression currently specified: {?}
  Report: OK.
->difference
  Expression currently specified: {(? \ nil)}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({?} \ nil)}
  Report: OK.
->divide
  Expression currently specified: {({(? / nil)} \ nil)}
  Report: OK.
->integer_constant(2)
  Expression currently specified: {({(2 / ?)} \ nil)}
  Report: OK.
->integer_constant(0)
  Expression currently specified: {({(2 / 0), ?} \ nil)}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ ?)}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {?})}
  Report: OK.
->add
  Expression currently specified: {({(2 / 0)} \ {(? + nil)})}
  Report: OK.
->divide
  Expression currently specified: {({(2 / 0)} \ {((? / nil) + nil)})}
  Report: OK.
->integer_constant(9)
  Expression currently specified: {({(2 / 0)} \ {((9 / ?) + nil)})}
  Report: OK.
->integer_constant(0)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + ?)})}
  Report: OK.
->integer_constant(132321)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321), ?})}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {?}}
  Report: OK.
->boolean_constant(True)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {?}}
  Report: OK.
->boolean_constant(False)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {?}}
  Report: OK.
->boolean_constant(True)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {?}}
  Report: OK.
->boolean_constant(True)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {?}}
  Report: OK.
->boolean_constant(False)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {?}}
  Report: OK.
->boolean_constant(False)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, ?}
  Report: OK.
->open_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, {?}}
  Report: OK.
->boolean_constant(False)
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, {False, ?}}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, {False}, ?}
  Report: OK.
->close_set_enumeration
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, {False}}
  Report: OK.
->evaluate
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, {False}}
  Report: Error (Divisor is zero).
->type_check
  Expression currently specified: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, {False}}
  Report: {({(2 / 0)} \ {((9 / 0) + 132321)}), {True}, {False}, {True}, {True}, {False}, {False}, {False}} is type-correct.
->reset
  Expression currently specified: ?
  Report: OK.
